bin/prod/update.sh

```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+
+if ps aux| grep -v grep |grep --quiet rclone; then
+  echo "rclone is running, not redeploying at this time"
+  echo "in future, consider adding delays/retries here"
+  exit 1
+else
+  echo rclone is not running, ok to deploy
+fi
+
+
+echo updating code...
+sudo bash -c "(cd /root/motuz && git pull)" # # TODO unhardcode motuz dir
```

Comment:
Let's move the git pull before line 3 to minimize the chance of race conditions.

Reply:
Alternatively extract lines 3-9 into a function and call it after the git pull again.

Reply:
I don't want to do the git pull if rclone is running, because then the code that is in the cloned repo will be out of sync with the containers that are running.
What race condition do you think could occur?

Reply:
`git clone` is not instant. A job might start during the potential 1-10 seconds of pull. I recommend that we check that no jobs are running twice. To keep the code DRY, extract the check to a function and call it before and after git clone?

Reply:
OK, I understand. Again, if we do a 'git pull' and then find that rclone is running afterwards, we will have a situation where the code in the local repository clone does not match what is in the containers. This is not a problem technically, but it is confusing. 

I can think of a potential solution but it is also a somewhat complex solution to what I hoped would be a simple action. The solution is as follows (it helps to know that the git repos is cloned into /root/motuz):

* Check to see if rclone is running. If it is, exit with error code 1. As you say, this code can be refactored into a function.
* Do a git `clone` (not a pull) into a temp directory
* Do another check to see if rclone is running. If it is, again exit, this time first deleting the temp directory from the previous step. 
* If rclone is not running, rename the current cloned repo to something else (motuz.bak), rename the temp directory we cloned into to /root/motuz. These two renames are atomic operations that should be instant (neither will be an mv across filesystems). So it is unlikely that a job started during that instant. 
* Now we can start the redeployment process, and in the background we can delete the old/renamed clone directory (motuz.bak). 

What do you think?